File: backend/src/main/java/com/optiroute/backend/entity/Customer.java
package com.optiroute.backend.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "customer")
public class Customer extends EntityUtils {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "external_id", length = 100)
    private String externalId;

    @Column(name = "external_source", length = 50)
    private String externalSource = "MANUAL";

    @Column(nullable = false, length = 255)
    private String name;

    @Column(length = 100)
    private String code;

    @Column(columnDefinition = "TEXT")
    private String address;

    @Column(length = 100)
    private String city;

    @Column(length = 100)
    private String country;

    // getters / setters

    public Long getId() {
        return id;
    }

    public String getExternalId() {
        return externalId;
    }

    public void setExternalId(String externalId) {
        this.externalId = externalId;
    }

    public String getExternalSource() {
        return externalSource;
    }

    public void setExternalSource(String externalSource) {
        this.externalSource = externalSource;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }
}

File: frontend/src/models/Customer.ts
export interface CustomerSummary {
    id: number;
    name: string;
    code: string | null;
    city: string | null;
}

export interface CustomerDetails {
    id: number;

    externalId: string | null;
    externalSource: string;

    name: string;
    code: string | null;

    address: string | null;
    city: string | null;
    country: string | null;

    createdAt: string;
    updatedAt: string;
}

export interface CustomerFormData {
    name: string;
    code: string | null;
    address: string | null;
    city: string | null;
    country: string | null;
}

export interface CustomerUpdateRequest extends CustomerFormData {
    externalId: string | null;
    externalSource: string;
}

export interface CustomerCreateRequest extends CustomerFormData {
    externalId: string | null;
    externalSource: string;
}